gptel.el

```diff
@@ -1567,8 +1567,9 @@ implementation, used by OpenAI-compatible APIs and Ollama."
                              ,@(if enum (list :enum (vconcat enum)))
                              ,@(cond
                                 ((equal type "object")
-                                 (list :parameters (plist-get arg :parameters)
-                                  :additionalProperties :json-false))
+                                 (list :properties (plist-get arg :properties)
+                                       :required (plist-get arg :required)
```

Comment:
If `:required` is missing in the object specification this becomes `:required nil` -- it should become `:required []` instead.

Reply:
Done https://github.com/karthink/gptel/pull/600/files#diff-58ae8200f2acac302967edbab7a33d6a0a6832a2b6dae6868375f4e5d1968795R1572